File: modules/x-engine-jsi-direct/android/jsi-direct/src/main/java/com/zkty/jsi/direct/JSI_direct.java
package com.zkty.jsi.direct;

import androidx.annotation.Nullable;

import com.zkty.engine.nativ.protocol.IDirectManager;
import com.zkty.nativ.core.NativeContext;
import com.zkty.nativ.direct.NativeDirect;
import com.zkty.nativ.jsi.bridge.CompletionHandler;

public class JSI_direct extends xengine_jsi_direct {

    private NativeDirect directors;

    @Override
    protected void afterAllJSIModuleInited() {
        if (directors instanceof NativeDirect)
            directors = (NativeDirect) NativeContext.sharedInstance().getModuleByProtocol(IDirectManager.class);
    }

    @Override
    public void _push(DirectPushDTO dto, CompletionHandler<Nullable> handler) {
        if (directors != null) {
            directors.push(dto.scheme, dto.host, dto.pathname, dto.fragment, dto.query, dto.params);
        }

    }

    @Override
    public void _back(DirectBackDTO dto, CompletionHandler<Nullable> handler) {
        if (directors != null) {
            directors.back(dto.scheme, null,  dto.fragment);
        }
    }


}

File: modules/x-engine-rn-direct/iOS/Class/JSI_direct.m
//
//  JSI_direct.h
//  ModuleApp
//
//  Created by zk on 2021/3/14.
//  Copyright © 2021 x-engine. All rights reserved.
//


#import "XENativeContext.h"
#import "JSI_direct.h"
#import "JSIContext.h"
#import "iDirectManager.h"
 

@interface JSI_direct ()
@property (nonatomic, strong)   id<iDirectManager>  directors;
@end
@implementation JSI_direct
JSI_MODULE(JSI_direct)

 
-(void)afterAllJSIModuleInited {
    self.directors = [[XENativeContext sharedInstance] getModuleByProtocol:@protocol(iDirectManager)];
}

- (void)_back:(DirectBackDTO *)dto complete:(void (^)(BOOL))completionHandler {
    [self.directors back:dto.scheme host:nil fragment:dto.fragment];
    completionHandler(YES);
  
}

- (void)_push:(DirectPushDTO *)dto complete:(void (^)(BOOL))completionHandler {  
    [self.directors push:dto.scheme host:dto.host pathname:dto.pathname fragment:dto.fragment query:dto.query params:dto.params];
    completionHandler(YES);
}

- (id)_back:(DirectBackDTO *)dto{
     [self.directors back:dto.scheme host:nil fragment:dto.fragment];
    return @"";
}


- (id)_push:(DirectPushDTO *)dto {
    [self.directors push:dto.scheme host:dto.host pathname:dto.pathname fragment:dto.fragment query:dto.query params:dto.params];
    return @"";
}
@end
